lib/libzfs/libzfs_util.c

```diff
@@ -1940,3 +1941,91 @@ zprop_iter(zprop_func func, void *cb, boolean_t show_all, boolean_t ordered,
 {
 	return (zprop_iter_common(func, cb, show_all, ordered, type));
 }
+
+static void
+nomem(void)
```

Comment:
I think that this code belongs in `zfs_main.c`, rather than libzfs.  If we keep it in libzfs, we should probably change it to not `exit()` or `printf()`, instead returning error codes and strings.

Reply:
I was a little surprised this wasn't centralized as I saw 'nomem()' and 'safe_malloc()' in various C files.  I think this is something that needs some tending to and try to DRY it everywhere.  I will leave this alone for now until consensus.

Reply:
It is centralized, at least to a degree.  See zfs_alloc() and no_memory() in this file.